staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go

```diff
@@ -0,0 +1,431 @@
+	cmd.MarkFlagRequired("image")
+	cmd.Flags().String("image-pull-policy", "", i18n.T("The image pull policy for the container. If left empty, this value will not be specified by the client and defaulted by the server"))
+	cmd.Flags().StringArray("env", []string{}, "Environment variables to set in the container.")
+	cmd.Flags().StringP("labels", "l", "", "Comma separated labels to apply to the pod(s). Will override previous values.")
+	cmd.Flags().BoolVarP(&opt.Interactive, "stdin", "i", opt.Interactive, "Keep stdin open on the container(s) in the pod, even if nothing is attached.")
+	cmd.Flags().BoolVarP(&opt.TTY, "tty", "t", opt.TTY, "Allocated a TTY for each container in the pod.")
+	cmd.Flags().BoolVar(&opt.Attach, "attach", opt.Attach, "If true, wait for the Pod to start running, and then attach to the Pod as if 'kubectl attach ...' were called.  Default false, unless '-i/--stdin' is set, in which case the default is true.")
+	cmd.Flags().Bool("command", false, "If true and extra arguments are present, use them as the 'command' field in the container, rather than the 'args' field which is the default.")
+	cmd.Flags().String("target", "", i18n.T("Target processes in this container name."))
+	cmd.Flags().BoolVar(&opt.Quiet, "quiet", opt.Quiet, "If true, suppress prompt messages.")
+}
+
+// Complete finishes run-time initialization of debug.Options.
+func (o *Options) Complete(f cmdutil.Factory, cmd *cobra.Command) error {
+	var err error
+
+	o.ArgsLenAtDash = cmd.ArgsLenAtDash()
+
+	attachFlag := cmd.Flags().Lookup("attach")
+	if !attachFlag.Changed && o.Interactive {
+		o.Attach = true
+	}
+
+	o.builder = f.NewBuilder()
+	clientset, err := f.KubernetesClientSet()
+	if err != nil {
+		return err
+	}
+	o.PodClient = clientset.CoreV1()
+
+	return nil
+}
+
+// Run executes a kubectl debug.
+func (o *Options) Run(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
+	names, containerArgs := args, []string{}
+	// If there is a dash and there are args after the dash, extract the command and args.
+	if o.ArgsLenAtDash >= 0 && len(args) > o.ArgsLenAtDash {
+		names, containerArgs = args[:o.ArgsLenAtDash], args[o.ArgsLenAtDash:]
+	}
+
+	if len(names) == 0 {
+		return cmdutil.UsageErrorf(cmd, "NAME is required for run")
+	}
+
+	// validate image name
+	imageName := o.Image
+	if imageName == "" {
+		return fmt.Errorf("--image is required")
+	}
+	if !reference.ReferenceRegexp.MatchString(imageName) {
+		return fmt.Errorf("Invalid image name %q: %v", imageName, reference.ErrReferenceInvalidFormat)
+	}
+
+	if o.TTY && !o.Interactive {
+		return cmdutil.UsageErrorf(cmd, "-i/--stdin is required for containers with -t/--tty=true")
+	}
+
+	namespace, _, err := f.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
+	}
+
+	if err := verifyImagePullPolicy(cmd); err != nil {
+		return err
+	}
+
+	generators := generateversioned.GeneratorFn("debug")
```

Comment:
I'm a big opponent of the generators we have and I'm slowly trying to get rid of them. I'd just create a method that generates the debug pod into this package as is. I'm finding the generators to be too limiting and not much useful, especially in this application.

Reply:
Great. I'm glad to hear that, I didn't like the generator either. I was just trying to be consistent. I've removed it and like the result much better.